_posts/2021-01-09-Julia-1.6-what-has-changed-since-1.0.md

```diff
@@ -0,0 +1,639 @@
+Since then all releases (except 1.6) have been timed releases.
+The release candidate is cut from the main branch every 3 months; and they are released after an [extensive round of additional checking](https://github.com/JuliaLang/julia/pull/35846), which often takes weeks to months.
+Julia 1.6 was soft-slated to be the new long-term-support (LTS) version that would have bug-fixes backported to it for the next few years.
+The current LTS is julia 1.0, which has now had 5 patch releases made.
+Since it was going to be supported for a long-time people wanted to make sure everything good got in; thus it was a feature release.
+The core developers have demurred on if 1.6 will actually be selected to be the new LTS (even if it wasn't it won't assend to being LTS til it stops being the current Stable).
+
+My impression now is that they feel like it has too many cool new things; and that a few things didn't quite make it in even with the extended release cycle.
+So it's looking likely to me that 1.7 will actually be the LTS; but that it might also be a feature release -- possibly this time a much shorter release period than usual.
+In practice I think for a lot of package maintainer's 1.6 will be a LTS, in that that is the oldest version they will make sure to continue to support.
+There has been too many cool new things (as this post will detail) to stay back to only 1.0 features.
+Already a lot of packages have dropped support for julia versions older than 1.3.
+
+
+This post is kind of a follow-up to my [Julia 1.0 release run-down]({{site.url}}//2018/06/01/Julia-Favourite-New-Things.html).
+But it's going to be even longer, as it is covering the last 5 releases since then; and I am not skipping the major new features.
+I am writing this not to break down release by release,
+but to highlight features that had you only used Julia 1.0, you wouldn't have seen.
+Full details can be found in the [NEWS.md](https://github.com/JuliaLang/julia/blob/backports-release-1.6/NEWS.md), and [HISTORY.md](https://github.com/JuliaLang/julia/blob/backports-release-1.6/HISTORY.md)
+
+
+## Front-end changes
+### Soft-scope in the REPL
+Julia 1.0 removed the notion of soft-scope from the language.
+I was very blasé about the change to for-loop bindings in my [1.0 release post](https://www.oxinabox.net/2018/06/01/Julia-Favourite-New-Things.html#for-loop-variable-binding-changes).
+Infact, I didn't even mention this particular change.
+It was [#19324](https://github.com/JuliaLang/julia/pull/19324) for reference
+
+This was undone in in Julia 1.5 with [#33864](https://github.com/JuliaLang/julia/pull/33864) for in the REPL only.
+Now in the REPL, assigning to a global variable within a for-loop actually assigns that variable, rather than shadowing it with a new variable in that scope.
+The same behavior outside the REPL, now gives a warning.
+
+Personally, this change never affected me because I never write for-loops at global scope that assign variables.
```

Comment:
```suggestion
Personally, this change never affected me because I never write for-loops that assign variables at global scope.
```

I feel that this reads a bit better, but it's subjective and I think both are correct